### PR TITLE
chore(ci): migrate to actions/java to use built-in sbt caching

### PR DESCRIPTION
**Pull request by ckipp01**
I'd like to see if this works. There were a couple coversations a while back about
using setup-scala vs setup-java and it seems that many repos have migrated to just
using setup-java since it does offer sbt support, has a ton of users, and also
built-in sbt caching which simplifies a few things in our actions.

If you're rather just leave it as is since it is working fine, no
worries, I just really like that the setup-java offers built-in caching
for sbt.

### CODE REVIEW

**Review by tgodzik — APPROVED** [review]
I think it's good to use the setup-java action instead of Olaf's one, since we don't really want to maintain it.

If it works, let's merge it!

**Comment by ckipp01** [comment]
Weird. So much more flaky.

**Comment by tgodzik** [comment]
> Weird. So much more flaky.

It doesn't look flaky actually, but rather uses a different JDK than we did previously.

**Comment by ckipp01** [comment]
> > Weird. So much more flaky.
> 
> It doesn't look flaky actually, but rather uses a different JDK than we did previously.

O duh, I moved to temurin because setup-java didn't list adopt as an option for 17. Lemme dig in to see if they actually do or what's the best approach.

**Comment by tgodzik** [comment]
Temurin isn't the new adopt? I was rather thinking that maybe jabba had a very old version of JDK 17 and the sources changed a bit.

**Comment by ckipp01** [comment]
> Temurin isn't the new adopt? I was rather thinking that maybe jabba had a very old version of JDK 17 and the sources changed a bit.

Yea from my understanding adopt is adoptium and releasing under temurin. 🤔 I'll need to dig in a bit more because I can change these to pass on 17, but they will then fail on 8.

**Comment by tgodzik** [comment]
> > Temurin isn't the new adopt? I was rather thinking that maybe jabba had a very old version of JDK 17 and the sources changed a bit.
> 
> Yea from my understanding adopt is adoptium and releasing under temurin. thinking I'll need to dig in a bit more because I can change these to pass on 17, but they will then fail on 8.

Don't we have already some different expected for Java 8 and 17 in most places?

**Comment by ckipp01** [comment]
> Don't we have already some different expected for Java 8 and 17 in most places?

Yup, that's what I'll use.

**Comment by ckipp01** [comment]
Alright ready to rock and roll if you want to give it another look over.

File: tests/unit/src/test/scala/tests/WorkspaceSymbolLspSuite.scala [hunk 1]
```diff
@@ -348,15 +365,25 @@ class WorkspaceSymbolLspSuite extends BaseLspSuite("workspace-symbol") {
...
+             |scala.jdk.FutureConverters
+             |scala.jdk.javaapi.FutureConverters
+             |scala.concurrent.impl.FutureConvertersImpl
+             |""".stripMargin
+        else
+          """|scala.concurrent.Future
+             |scala.concurrent.Future
+             |scala.sys.process.ProcessImpl#Future
+             |scala.jdk.FutureConverters.FutureOps
+             |java.io.ObjectStreamClass#EntryFuture
```
Comment (tgodzik) [inline tests/unit/src/test/scala/tests/WorkspaceSymbolLspSuite.scala:382]: Weirdly enough this seems to be available still in JDK 17 :thinking:

**Review by tgodzik — APPROVED** [review]
Looks good, let merge it after the release.

**Pull request merged**